### PR TITLE
object-specific sampler learning

### PR DESCRIPTION
**Pull request by tsilver-bdai**
PR to include object-specific sampler learning change on predicators OG.

Also includes:
* change that fixes grasp samplers (specifically, rotation constraint was badly broken for cup sampler)
* fixes to the place sampler visualization script given this object-specific change

### CODE REVIEW

**Comment by tsilver-bdai** [comment]
I want to be careful because there's some stuff here that I reverted upstream (like the feature selection for the ball cup env), so after upstream merges, I will create a fresh PR to merge just those changes, and then update this PR on top of that one.

**Comment by nkumar-bdai** [comment]
yup yup - sounds good!

File: predicators/spot_utils/perception/object_specific_grasp_selection.py [hunk 1]
```diff
@@ -65,9 +65,7 @@ def _get_ball_grasp_pixel(
     mask = seg_bb.mask
     pixels_in_mask = np.where(mask)
     pixel = (pixels_in_mask[1][-1], pixels_in_mask[0][-1])
-    # Force a forward top-down grasp.
-    rot_quat = math_helpers.Quat.from_pitch(np.pi / 2)
-    return pixel, rot_quat
+    return pixel, None
```
Comment (tsilver-bdai) [inline predicators/spot_utils/perception/object_specific_grasp_selection.py:68]: just double checking that this change is intentional @NishanthJKumar

**Comment by tsilver-bdai** [comment]
@NishanthJKumar approve and merge when ready

File: predicators/spot_utils/perception/object_specific_grasp_selection.py (hunk 1 above)
Comment (nkumar-bdai) [reply predicators/spot_utils/perception/object_specific_grasp_selection.py:68]: intentional!

**Review by nkumar-bdai — APPROVED** [review]
yayyy! hopefully no more significant bugs lurking for us to get to a reliable ball-and-cup demo

**Pull request merged**